File: matiasmasso/Mat.NET/DT2/DTOBasket.vb
Public Class DTOBasket
    Property customer As Guid
    Property customerUrl As String
    Property addressLines As List(Of String)
    Property concept As String
    Property totjunt As Boolean
    Property fchmin As String
    Property obs As String
    Property includePendingOrders As Boolean
    Property lines As List(Of DTOBasketLine)
    Property suma As Decimal
    Property sumaFormatted As String
    Property Id As Integer
    Property mailConfirmation As Boolean
    Property Catalog As DTOBasketCatalog
End Class

Public Class DTOBasketLine
    Property qty As Integer
    Property availableStock As Integer
    Property pending As Integer
    Property sku As Guid
    Property category As Guid
    Property brand As Guid
    Property url As String
    Property nom As String
    Property price As Decimal
    Property dto As Decimal
    Property Amount As Decimal
    Property priceFormatted As String
    Property amountFormatted As String

    Shared Function Factory(oItem As DTOPurchaseOrderItem) As DTOBasketLine
        Dim retval As New DTOBasketLine
        With retval
            .sku = oItem.Sku.Guid
            .category = oItem.Sku.Category.Guid
            .brand = oItem.Sku.Category.Brand.Guid
            .qty = oItem.Qty
            .nom = oItem.Sku.NomLlarg
            .price = oItem.Price.Eur
            .priceFormatted = DTOAmt.CurFormatted(oItem.Price)
            .dto = oItem.Dto
            .availableStock = Math.Min(oItem.Qty, oItem.Sku.Stock)
            .pending = oItem.Qty - .availableStock
            Dim oAmt = oItem.Amount
            .Amount = oAmt.Eur
            .amountFormatted = DTOAmt.CurFormatted(oAmt)
        End With
        Return retval
    End Function
End Class

File: matiasmasso/Mat.NET/DTO/DTOBasket.cs
using System;
using System.Collections.Generic;

namespace DTO
{
    public class DTOBasket
    {
        public Guid customer { get; set; }
        public string customerUrl { get; set; }
        public List<string> addressLines { get; set; }
        public string concept { get; set; }
        public bool totjunt { get; set; }
        public string fchmin { get; set; }
        public string obs { get; set; }
        public bool includePendingOrders { get; set; }
        public List<DTOBasketLine> lines { get; set; }
        public decimal suma { get; set; }
        public string sumaFormatted { get; set; }
        public int Id { get; set; }
        public bool mailConfirmation { get; set; }
        public DTOBasketCatalog Catalog { get; set; }
    }

    public class DTOBasketLine
    {
        public int qty { get; set; }
        public int availableStock { get; set; }
        public int pending { get; set; }
        public Guid sku { get; set; }
        public Guid category { get; set; }
        public Guid brand { get; set; }
        public string url { get; set; }
        public string nom { get; set; }
        public decimal price { get; set; }
        public decimal dto { get; set; }
        public decimal Amount { get; set; }
        public string priceFormatted { get; set; }
        public string amountFormatted { get; set; }

        public static DTOBasketLine Factory(DTOPurchaseOrderItem oItem)
        {
            DTOBasketLine retval = new DTOBasketLine();
            {
                var withBlock = retval;
                withBlock.sku = oItem.Sku.Guid;
                withBlock.category = oItem.Sku.Category.Guid;
                withBlock.brand = oItem.Sku.Category.Brand.Guid;
                withBlock.qty = oItem.Qty;
                withBlock.nom = oItem.Sku.NomLlarg.Esp;
                withBlock.price = oItem.Price.Eur;
                withBlock.priceFormatted = DTOAmt.CurFormatted(oItem.Price);
                withBlock.dto = oItem.Dto;
                withBlock.availableStock = Math.Min(oItem.Qty, oItem.Sku.Stock);
                withBlock.pending = oItem.Qty - withBlock.availableStock;
                var oAmt = oItem.Amount();
                withBlock.Amount = oAmt.Eur;
                withBlock.amountFormatted = DTOAmt.CurFormatted(oAmt);
            }
            return retval;
        }
    }
}
